models/migrations/v999.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"xorm.io/xorm"
+)
+
+func creatUserBadgesTable(x *xorm.Engine) error {
+	type Badge struct {
+		ID          int64 `xorm:"pk autoincr"`
+		Description string
+		ImageURL    string
+	}
+
+	type userBadge struct {
+		ID      int64 `xorm:"pk autoincr"`
+		BadgeID int64
```

Comment:
Add an index is better.

Reply:
I would strongly suggest that adding a single index on `badge_id` is going to be incorrect and slow things down once this table gets larger.

The index would likely be chosen in favour of an index on `user_id` in any query involving both params even though `user_id` would be the better one.

This field should only be part of an composite index.